dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -324,82 +417,111 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
     *   d.select( d('a), d('a)+d('b), ... )
     * }}}
     */
-  def select[A: TypedEncoder, B: TypedEncoder](
+  def select[A, B](
     ca: TypedColumn[T, A],
     cb: TypedColumn[T, B]
-  ): TypedDataset[(A, B)] = selectMany(ca, cb)
+  ): TypedDataset[(A, B)] = {
+    implicit val ea = ca.uencoder; implicit val eb = cb.uencoder;
+
+    selectMany(ca, cb)
+  }
 
   /** Type-safe projection from type T to Tuple3[A,B,...]
     * {{{
     *   d.select( d('a), d('a)+d('b), ... )
     * }}}
     */
-  def select[A: TypedEncoder, B: TypedEncoder, C: TypedEncoder](
+  def select[A, B, C](
     ca: TypedColumn[T, A],
     cb: TypedColumn[T, B],
     cc: TypedColumn[T, C]
-  ): TypedDataset[(A, B, C)] = selectMany(ca, cb, cc)
+  ): TypedDataset[(A, B, C)] = {
+    implicit val ea = ca.uencoder; implicit val eb = cb.uencoder; implicit val ec = cc.uencoder;
```

Comment:
Maybe this would read better as `ìmplicit val (ea, eb, ec) = (ca.uencoder, cb.uencoder, cc.uencoder)`?